apps/multisig/src/layouts/Settings/DescriptionForm.tsx

```diff
@@ -0,0 +1,122 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import { SettingsInfoRow } from './InfoRow'
+import { Button } from '@components/ui/button'
+import { Edit, X } from 'lucide-react'
+import { Input } from '@components/ui/input'
+import Modal from '@components/Modal'
+import { useMutation } from '@apollo/client'
+import { gql } from 'graphql-tag'
+import { useToast } from '@components/ui/use-toast'
+import { organisationsState } from '@domains/offchain-data'
+import { useSetRecoilState } from 'recoil'
+
+type Props = {
+  description: string
+  editable?: boolean
+  teamId: string
+}
+
+const UPDATE_VAULT_DESCRIPTION = gql`
+  mutation UpdateTeamDescription($teamId: uuid!, $description: String!) {
+    update_team_by_pk(pk_columns: { id: $teamId }, _set: { description: $description }) {
+      id
+      description
+    }
+  }
+`
+export const DescriptionForm: React.FC<Props> = ({ editable, description, teamId }) => {
+  const [editing, setEditing] = useState(false)
+  const [newDescription, setNewDescription] = useState(description)
+  const setOrganisations = useSetRecoilState(organisationsState)
+  const cachedDescription = useRef(description)
+  const { toast } = useToast()
+  const [mutate, { loading }] = useMutation<{ update_team_by_pk: { id: string; description: string } }>(
+    UPDATE_VAULT_DESCRIPTION,
+    {
+      variables: {
+        teamId,
+        description: newDescription,
+      },
+      onCompleted: data => {
+        if (data.update_team_by_pk && data.update_team_by_pk.id === teamId) {
+          toast({ title: 'Multisig description updated' })
+          setEditing(false)
+          setOrganisations(prev => {
+            const newOrgs = prev.map(org => {
+              const teams = org.teams.map(team =>
+                team.id === teamId ? { ...team, description: newDescription } : team
+              )
+              return { ...org, teams }
+            })
+            return newOrgs
+          })
+        }
+      },
+    }
+  )
+
+  useEffect(() => {
+    if (cachedDescription.current !== description || !editing) {
+      setNewDescription(description)
+      cachedDescription.current = description
+    }
+  }, [editing, description])
```

Comment:
I think dont need cachedDescription, we can refer to the description prop as the cached. I haven't test this but I imagine the UX would be:
1. user clicks the Edit
2. Prefills description if any
3. user change description but exit without saving
4. this effect gets invoked and reset newDescription to description
5. user comes back and sees the description unchanged from existing description in DB
```suggestion
  useEffect(() => {
    if (!editing && newDescription !== description) {
      setNewDescription(description)
    }
  }, [editing, description])
```